src/SensioLabs/JobBoardBundle/Service/JobFetcher.php

```
@@ -0,0 +1,48 @@
+<?php
+
+
+namespace SensioLabs\JobBoardBundle\Service;
+
+use Doctrine\ORM\EntityManager;
+use Pagerfanta\Adapter\DoctrineORMAdapter;
+use Pagerfanta\Pagerfanta;
+
+class JobFetcher
+{
+    /** @var \Doctrine\ORM\EntityManager */
+    private $manager;
+
+    /**
+     * @param \Doctrine\ORM\EntityManager $manager
+     */
+    public function __construct(EntityManager $manager)
+    {
+        $this->manager = $manager;
+    }
+
+    /**
+     * @param int $page
+     *
+     * @return array|null|\Traversable
+     */
+    public function fetch($page)
+    {
+        $queryBuilder = $this
+            ->manager
+            ->getRepository('SensioLabsJobBoardBundle:Job')
+            ->getQueryBuilderForPagination();
+
+        $adapter = new DoctrineORMAdapter($queryBuilder);
+        $pagerfanta = new Pagerfanta($adapter);
+        $pagerfanta->setMaxPerPage(10);
+
+        if ($page <= $pagerfanta->getNbPages()) {
+            $pagerfanta->setCurrentPage($page);
+            $results = $pagerfanta->getCurrentPageResults();
+        } else {
+            $results = null;
```

Hi, i agree with @ogizanagi with :
* you should throw a client error when you're asking a none existing resource (but i preferer a 400 badRequestHttpException rather than a NotFound one)
* You can throw a corresponding exception in your repository method. the most adapted exception is : https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/NoResultException.php

Why we're talking about JS error ? :confused: